src/gpu/util.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+// the atomic update overloads for double are in the root namespace
+// so that they match the namespace of the CUDA builtin equivalents
+// for 32 bit float and 32/64 bit int
+/*
+__device__
+inline double atomicAdd(double* address, double val)
+{
+    using I = unsigned long long int;
+    I* address_as_ull = (I*)address;
+    I old = *address_as_ull, assumed;
+    do {
+        assumed = old;
+        old = atomicCAS(
+                address_as_ull,
+                assumed,
+                __double_as_longlong(val + __longlong_as_double(assumed))
+        );
+    } while (assumed != old);
+    return __longlong_as_double(old);
+}
+
+__device__
+static double atomicSub(double* address, double val)
+{
+    return atomicAdd(address, -val);
+}
+
+__device__
+static float atomicSub(float* address, float val)
+{
+    return atomicAdd(address, -val);
+}
+*/
+
+namespace nest {
+namespace mc {
+namespace util {
+namespace gpu {
+
+
+
```

Comment:
This is a mess! Before Pascal GPUs, there was no `atomic_add` for `double`, so we had to roll our own with `CAS`. Pascal now natively supports atomic operations for `double`, so these are defined by the compiler. Unfortunately they are defined as "magic functions" inside the cuda 8 compiler, and are present even when compiling for non-Pascal architecture... mess ensues.

I have a hacked solution where `atomic_add` was defined inside the namespace of each mechanism, and forgot to remove this file.

I have removed it now.